File: src/MXHRM.Application/Common/Realtime/RealtimeMessage.cs
namespace MXHRM.Application.Common.Realtime;

public sealed class RealtimeMessage
{
    public long? NotificationId { get; set; }

    public string? Key { get; set; }

    public string Type { get; set; } = string.Empty;

    public string? Title { get; set; }

    public string? Message { get; set; }

    public string? Tone { get; set; }

    public string? Route { get; set; }

    public object? Data { get; set; }

    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
}

File: client/mxhrm-web/src/app/core/realtime/realtime-message.ts
export interface RealtimeMessage<TData = unknown> {
    notificationId?: number | null;
    key?: string | null;
    type: string;
    title?: string | null;
    message?: string | null;
    tone?: string | null;
    route?: string | null;
    data?: TData | null;
    createdAtUtc: string;
}